hacep/src/main/java/org/kie/hacep/core/infra/consumer/DefaultKafkaConsumer.java

```diff
@@ -58,19 +56,17 @@
 
     private Logger logger = LoggerFactory.getLogger(DefaultKafkaConsumer.class);
     private Map<TopicPartition, OffsetAndMetadata> offsetsEvents = new HashMap<>();
-    private org.apache.kafka.clients.consumer.Consumer<String, T> kafkaConsumer, kafkaSecondaryConsumer;
+    private Consumer<String, T> kafkaConsumer, kafkaSecondaryConsumer;
     private DroolsConsumerHandler consumerHandler;
-    private volatile State currentState;
     private volatile String processingKey = "";
     private volatile long processingKeyOffset, lastProcessedControlOffset, lastProcessedEventOffset;
-    private volatile boolean leader = false;
-    private volatile boolean started = false;
-    private volatile boolean processingLeader, processingReplica = false;
-    private volatile boolean pollingEvents, pollingControl = true;
+    private volatile boolean started, exit = false;
```

Comment:
Can you please explain better the lifecycle of `started` variable?
Do we need a variable to pause/resume consuming data?
From what I can understand it could lead to a 100% CPU while true because if `started` is `false` and execution reach this point
`while (!exit) {
      consume(durationMillis);
}`
then `consume(..)` will skip the whole content in a while true.

If `exit = false` and `started = true` cannot happen then probably it is enough to have a single boolean for the whole status.

Reply:
Started is the variable to starts the consume after all the mandatory stuffs are done at the end of enableConsumeAndStartLoop, only one is not enough because we use start = false calling stopConsume in the restart if one pod change from replica to leader or reverse, exit is to exit completely from the loop because we are stopping the pod to kill.